### ISSUE TITLE
OperationalError: Database or disk is full on GlobaLeaks  Despite Sufficient Disk Space

### ISSUE THREAD

**Issue by willskymaker** [globaleaks/GlobaLeaks#4170]
### What version of GlobaLeaks are you using?

v4.14.8

### What browser(s) are you seeing the problem on?

All

### What operating system(s) are you seeing the problem on?

Linux

### Describe the issue

We are encountering a persistent OperationalError on our GlobaLeaks installation, specifically version 4.14.8. The error indicates that the "database or disk is full," despite confirming that there is sufficient disk space available on the server.

Error Traceback:

sqlalchemy.exc.OperationalError Wraps a DB-API OperationalError.

Traceback (most recent call last):

  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(

  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)

sqlite3.OperationalError: database or disk is full

...

sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) database or disk is full
[SQL: INSERT INTO mail (id, tid, creation_date, address, subject, body) VALUES (?, ?, ?, ?, ?, ?)]
[parameters: ('5daffd23-9c9e-4a02-833f-dcddc6d87fd8', 1, '2024-08-17 22:05:57.571743', 'whistleblowing@plu***.eu', 'GlobaLeaks Exception', ...]
(Background on this error at: https://sqlalche.me/e/14/e3q8)

Environment:

GlobaLeaks Version: 4.14.8
Host: segnalazioni.plu***.it (via Tor and HTTPS)
Operating System: Debian 12
Database: SQLite
Steps to Reproduce:

Operate the GlobaLeaks platform under normal conditions.
The system intermittently triggers the above OperationalError indicating that the database or disk is full.
What we have tried:

Verified that there is sufficient disk space on the server.
Checked file system quotas and disk usage.
Considered the possibility of corruption but found no evidence.

Expected Behavior:

The system should continue to function without triggering this error if sufficient disk space is available.
Potential Cause: Given that the current GlobaLeaks version is 4.14.8, and the latest stable release is 5.0.2, we suspect that this issue might be related to an outdated version of the software. It is possible that this issue has been addressed in subsequent updates.

Request:

We seek confirmation on whether this issue is resolved in later versions.
Any suggestions on mitigating this error while we prepare to upgrade to the latest version would be appreciated.
Next Steps:

We plan to upgrade to version 5.0.2 but would like to understand if this issue is recognized and any recommended steps before proceeding with the upgrade.

![issue](https://github.com/user-attachments/assets/044f4b7d-c2a3-4c6b-9f39-4370249fe493)

**Comment by willskymaker** [comment]
Thank you for your prompt response.

To help you replicate the issue, here are the detailed steps and additional context about our setup:

Replication Steps:

Environment Setup:
Install GlobaLeaks version 4.14.8 on a server with sufficient SSD space (we have over 200GB free on our system) and 1TB of Backup Storage.
The server is hosted on a cloud environment with no unusual restrictions.

Normal Operation:
Operate the GlobaLeaks platform under typical conditions, including sending notifications and handling submissions.
Error Occurrence:
The error usually appears when the platform tries to insert a new record into the database (e.g., sending an email notification).
No specific trigger action leads to the error consistently; it happens during normal use.

Additional Details:

We observed this issue even though there is no apparent disk space or quota limitation, which suggests it may not be directly related to actual storage availability.
The system's current settings do not impose specific restrictions on SQLite or GlobaLeaks that would cause this error under normal circumstances.

Questions:

Are there any known issues or changes related to database handling between version 4.14.8 and the current stable release 5.0.2 that could address this problem?
Are there specific logs or further diagnostics we should enable to provide more useful data?
Next Steps:

We are preparing to upgrade to the latest version but would greatly appreciate any guidance on how to manage this error in the meantime.
We hope this helps you understand the context and replicate the issue. Let us know if there are any additional tests or configurations you would like us to try.

Thank you for your support!

**Comment by evilaliv3** [comment]
Updating to 5.0.2 is recommended since its a version that includes the update of the client from AngularJs to Angular that is the stable library, but actually that version does not include changes in the database handling.

Such an error seems to me related to a possible disk problem; Which filesystem are you using? Have you recently changed the way you handle the /var/globaleaks directory? (maybe is a remote partition?)

**Comment by willskymaker** [comment]
Thank you for your prompt response and support.

After conducting further tests, I identified the cause of the issue. The problem was related to the SMTP server configuration. I changed the SMTP server settings and noticed that the recipient field in the configuration panel was empty, likely due to some internal check in the code related to the presence of a working SMTP server.

Once I updated the SMTP server settings and reinserted the recipient, the system resumed normal operation, and no further errors were encountered. 

Everything is now functioning correctly, and the platform is stable.

I appreciate your assistance, and this issue can now be considered resolved. 

Thanks again for your help!
